### PR TITLE
Add 'New password must differ from the old one' check

### LINKED ISSUE #231: Password change
Should display the message  "The new password cannot be same as the old one" when new password and older password is same.

### PR DESCRIPTION
**Pull request by nathalier**
To esolve the issue ChangePasswordForm was redifined, namely clean_password1() method added.

Fixes #231.

### CODE REVIEW

File: systers_portal/users/forms.py [hunk 1]
```diff
@@ -36,3 +38,16 @@ class SystersUserForm(forms.ModelForm):
...
+
+
+class SystersChangePasswordForm(ChangePasswordForm):
+    # The clean_password1() method which checks that new password differs from the old one
+    # is added to allauth ChangePasswordForm class.
+    # So ChangePasswordForm class is derived.
+    def __init__(self, *args, **kwargs):
+        super(SystersChangePasswordForm, self).__init__(*args, **kwargs)
+
+    def clean_password1(self):
```
Comment (tapasweni-pathak) [inline systers_portal/users/forms.py]: what about ```check_new_password``` or do you have a better suggestion?

**Comment by sa7i** [comment]
@tapasweni-pathak I think this can be merged. From [here](https://github.com/pennersr/django-allauth/blob/25fe632acf12571ae2ac9e692e8890019d5a6e7b/allauth/account/forms.py#L457) The new password's field is password1 and from [this](https://stackoverflow.com/a/44742997/8800466) "One can validate only that specific field in the clean_{fieldname} function."

**Comment by tapasweni-pathak** [comment]
@sa7i did not merge because of the field name, could change and merge, thanks for the ping.

File: systers_portal/users/forms.py [hunk 2]
```diff
@@ -36,3 +38,16 @@ class SystersUserForm(forms.ModelForm):
...
+        if self.cleaned_data["password1"] == self.cleaned_data.get("oldpassword", None):
```
Comment (tapasweni-pathak) [inline systers_portal/users/forms.py]: Added a minor commit to change few variable names here.

File: systers_portal/users/forms.py [hunk 3]
```diff
@@ -36,3 +38,16 @@ class SystersUserForm(forms.ModelForm):
...
+            raise ValidationError("New password must differ from the old one.")
+        return self.cleaned_data["password1"]
```
Comment (tapasweni-pathak) [inline systers_portal/users/forms.py]: Added an extra commit to change few variable names here and merged w/ 3d870165ad0b93e55c084a2e463bede642ec6ff3.

**Pull request merged**